maps/maps_test.go

```diff
@@ -65,3 +65,63 @@ func TestStateMap_HasState(t *testing.T) {
 		})
 	}
 }
+
+func TestMetadata_Update(t *testing.T) {
+	type args struct {
+		key   string
+		value string
+	}
+	tests := []struct {
+		name string
+		m    Metadata
+		args args
+	}{
+		{
+			name: "success - update key and value",
+			m:    map[string]string{"key1": "value2"},
+			args: args{key: "key1", value: "value1"},
+		},
+		{
+			name: "success - update key and value with nil map",
+			m:    nil,
+			args: args{key: "key1", value: "value1"},
+		},
```

Comment:
we might wanna consider adding tests for adding a new key and value to an existing map